tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/StackTraceLeak/StackTraceLeakIntegrationCommon.cs

```diff
@@ -0,0 +1,40 @@
+// <copyright file="StackTraceLeakIntegrationCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Iast;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.StackTraceLeak;
+
+#nullable enable
+internal static class StackTraceLeakIntegrationCommon
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(StackTraceLeakIntegrationCommon));
+
+    internal static CallTargetState OnExceptionLeak(IntegrationId integrationId, Exception exception)
+    {
+        if (!Tracer.Instance.Settings.IsIntegrationEnabled(integrationId))
```

Comment:
It's checked later in IASTModule. Still, this piece of code should not be reached if IAST is disabled because of the instrumentation category.

Reply:
> this piece of code should not be reached if IAST is disabled because of the instrumentation category

Good point, I had forgotten that we don't even rejit these if IAST isn't enabled 🙂